src/source.cc

```diff
@@ -68,6 +75,11 @@ std::set<cyclus::BidPortfolio<cyclus::Material>::Ptr> Source::GetMatlBids(
   if (max_qty < cyclus::eps()) {
     return ports;
   } else if (commod_requests.count(outcommod) == 0) {
+    // if there is no demand for commodity, source dumps all throughput to buffer.
```

Comment:
It should actually create the material here. It's not enough to just keep track of the amount that should be on hand. If you're storing it, it should be subject to decay, etc., like any other material in the simulation.